sdk/include/opentelemetry/sdk/metrics/export/metric_producer.h

```diff
@@ -31,8 +31,21 @@ namespace metrics
  */
 struct ScopeMetrics
 {
-  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_;
+  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_ = nullptr;
   std::vector<MetricData> metric_data_;
+
+  template <class ScopePtr, class MetricDataType>
+  inline ScopeMetrics(ScopePtr &&scope, MetricDataType &&metric)
+      : scope_{std::move(scope)}, metric_data_{std::move(metric)}
+  {}
+
+  inline ScopeMetrics() {}
+  inline ScopeMetrics(const ScopeMetrics &) = default;
+  inline ScopeMetrics(ScopeMetrics &&)      = default;
+
+  inline ScopeMetrics &operator=(const ScopeMetrics &) = default;
+
+  inline ScopeMetrics &operator=(ScopeMetrics &&) = default;
 };
 
 struct ResourceMetrics
```

Comment:
It didn't help me, but you might want to add the changes for `resource_` as well.